### ISSUE TITLE
Nothing is happening when deploying from localhost to Netlify 

### ISSUE THREAD

**Issue by sebastiandammark** [Mulliman/xStatic-for-Umbraco#32]
When deploying from localhost to Netlify nothing is happening.

The deploy is pickup in Netlify, but all I see is a spinner.
The deploy itself says **Uploading**, but nothing happens.

Locally I can see that **Deploy duration** is empty, which I find a bit odd.
But otherwise everything looks right.

<img width="710" alt="Screenshot 2024-08-30 at 22 09 08" src="https://github.com/user-attachments/assets/a4847265-b734-4622-b64d-6ee019373505">
<img width="389" alt="Screenshot 2024-08-30 at 22 07 02" src="https://github.com/user-attachments/assets/2857a239-81bc-41b9-874e-72e980c20956">
<img width="830" alt="Screenshot 2024-08-30 at 22 06 44" src="https://github.com/user-attachments/assets/9accb0fc-90e6-490c-9c8c-7aedec2c892a">

**Comment by Mulliman** [comment]
Hi

I can't say I've seen this before, not sure if it's a temporary issue with Netlify, and issue will the files to deploy, or a bug in the xStatic code.

Have you cancelled that build and tried again? did the same happen?

Do the files in App_Data\xStatic\output\{id} look correct?

Are there any errors in the logs?

Did you create the netlify site using git or a folder with a single index.html file?

**Comment by Mulliman** [comment]
I can see that last build duration looks to be empty, which also seems a bit suspicious.

**Comment by sebastiandammark** [comment]
Hmmm ... all local files are generated so I have no idea why the Build duration is unspecified.
I've attached screendumps and the latest tracelog.

[UmbracoTraceLog.SebastiansMBP2.20240831.json](https://github.com/user-attachments/files/16824729/UmbracoTraceLog.SebastiansMBP2.20240831.json)

<img width="559" alt="Screenshot 2024-08-31 at 14 27 08" src="https://github.com/user-attachments/assets/b5e24488-a728-43fc-829a-3ec35b2b374c">

<img width="315" alt="Screenshot 2024-08-31 at 14 26 25" src="https://github.com/user-attachments/assets/fe90f493-2e6d-4edc-87ea-d6bad6463a10">

**Comment by Mulliman** [comment]
I can't see anything in the logs there around the deployment stage.

Do you have an error message in the javascript console, or more likely in the network tab of dev tools?

In v13 you should be able to just call this URL in the browser (changed in v14 now), maybe see if you get an error doing that: /umbraco/backoffice/xstatic/Deploy/DeployStaticSite/?staticSiteId={your id}

**Comment by sebastiandammark** [comment]
I have no errors in the console.

But if I open the link directly in the browser I get something looking like this.
```
)]}',
{"Type":null,"Title":null,"Status":417,"Detail":null,"Instance":null,"Extensions":{"traceId":"00-e572f1c69b65469f23dc0a624db57f88-9e66337f11e4a64e-00"}}
```

Which doesn't look quite right to me.

**Comment by Mulliman** [comment]
I just relooked at your screenshot. Are you using a mac, and could that make a difference?

**Comment by Mulliman** [comment]
The bad news is that I've just created a completely blank 13.4.1 Umbraco install with xStatic 13.0.1, went through the setup, and could deploy to Netlify first time without issue.

There is a non-zero chance that Mac could be the issue here, and I have no way to help with it.

**Comment by Mulliman** [comment]
One thing potentially worth trying is the following command

`dotnet new install Umbraco.Templates::14.2.0 --force && dotnet new umbraco --force -n "FourteenNetlifyNuget141" --friendly-name "Administrator" --email "admin@admin.com" --password "1234567890" --development-database-type SQLite && dotnet add "FourteenNetlifyNuget141" package Umbraco.TheStarterKit && dotnet add "FourteenNetlifyNuget141" package xStatic --version 14.1.0 && dotnet add "FourteenNetlifyNuget141" package xStatic.Netlify --version 14.1.0 && dotnet run --project "FourteenNetlifyNuget141"`

In 14.1 I've automated the creation of Netlify sites and I've also started using a Netlify NuGet package rather than using a deprecated web client.

- run the command in powershell 7
- login with admin@admin.com / 1234567890
- on Deployment Targets click `Automatically Create New Deployment Target` and follow instructions
- configure a site using this
- build
- deploy

You don't need to make any other set up changes or coding so this is probably the simplest way to check if a mac issue or a v13 issue.

**Comment by sebastiandammark** [comment]
Hi Sam,

The script is running without any issues, but there is no sign of xStatic in the backoffice.
It seems though that the packages have been installed.
![Screenshot 2024-09-06 at 08 51 47](https://github.com/user-attachments/assets/9a3a0cc0-85c0-471f-97d5-10f0d32f99f8)
![Screenshot 2024-09-06 at 08 52 15](https://github.com/user-attachments/assets/50a4c156-9483-4575-82e1-5536cde0de65)

**Comment by Mulliman** [comment]
We really aren't having much luck with this on your machine.

It should've created two new roles in User Groups, and assigned them to the root user. You may need to manually assign the new roles to your user for some reason, but for me it goes through automatically.

**Comment by sebastiandammark** [comment]
Hmmmm ... The roles are already assigned.
![image](https://github.com/user-attachments/assets/7eeeb7b2-a21d-4677-a7c6-c495f74cc1ed)

But it looks like the xStatic section is missing
![image](https://github.com/user-attachments/assets/df3a2f5b-758e-49fc-835a-dc9c1c8fcb03)

**Comment by Mulliman** [comment]
I've just run that exact script another time, and I can see the xStatic section right away.

I did notice that the starterkit styles didn't load first time, so I stopped the running command, went to the containing directory, and used `dontent run` and then the starter kit styles started working.

Maybe there's an issue inside Umbraco that stops files being served on the first run, and I get that with the starterkit but you get it with xStatic.

**Comment by sebastiandammark** [comment]
Yeah I had the same issue with the Starter Kit, but I simply can't get access to the xStatic section.
It's simply not visible anywhere.

I've tried to execute the command twice, restarting, deleting the user groups, restarting, dotnet clean, restarting.
Nothing works.

**Comment by Mulliman** [comment]
I've run this script at least half a dozen times in the last couple of days, and haven't had an issue with it. I've also pushed loads of changes up to Netlify also without issue. I can't think of any reason why you'd be having so many issue with both v13 and v14.

**Comment by sebastiandammark** [comment]
I'll see if I can get a hold of a scrappy Windows machine :)

**Comment by Mulliman** [comment]
I've released a new video today around the new 14.1 scripted installation, and on the thread have asked if another Mac user would give it a go. Fingers crossed someone will also give it a go.

**Comment by Mulliman** [comment]
Another Mac user has been able to replicate the v14 issue that you have. I'll see if I can find a way to replicate.

**Comment by Mulliman** [comment]
I've lowercase everything I can find in 14.2.0-alpha, just setting up a linux VM to test it

**Comment by Mulliman** [comment]
Tested in Linux, still no luck :/

**Comment by Mulliman** [comment]
Ok, some progress.

xStatic now shows in Linux for me after some fixes, and I can now replicate the Netlify issue. I think I do some path manipulation that looks pretty windows specific.

**Comment by Mulliman** [comment]
I've managed to narrow this down to a SSL trust issue on Linux (so hopefully in the same on Mac).

The two options are to:

1: Properly set up and trust certificates on the website host
2: Use a setting (that I will need to update) that trusts all HTTP traffic that the HttpClient that xStatic creates.

1 is safer, but for local dev 2 should be fine as it only calls localhost.

**Comment by Mulliman** [comment]
This now works on Linux

```
dotnet new install Umbraco.Templates --force && dotnet new umbraco --force -n "FixedOnLinux" --friendly-name "Administrator" --email "admin@admin.com" --password "1234567890" --development-database-type SQLite && dotnet add "FixedOnLinux" package clean && dotnet add "FixedOnLinux" package xStatic --version 14.2.0-beta2 && dotnet add "FixedOnLinux" package xStatic.Netlify --version 14.2.0-beta2 && dotnet run --project "FixedOnLinux"
```

With this added to program.cs

using XStatic.Core.App;

builder.Services.AddXStatic()
    .Automatic()
    .UseSecureUserGroups("admin@admin.com")
    .TrustUnsafeSslConnectionWhenGenerating()
    .Build();
